### ISSUE TITLE
Be careful if you want to use this library in production. Mutations may be executed more than once.

### ISSUE THREAD

**Issue by shawnlewis** [helfer/apollo-link-serialize#12]
If you have some other middleware that subscribes to its observable, this library will make every "serialized" mutation happen twice.

We recently added a middleware like this:
```
const noErrorLink = new ApolloLink((operation, forward) => {
  const observable = forward!(operation);
  observable.subscribe({
    complete: () => dispatch(resetErrorPortal()),
    error: () => {},
  });
  return observable;
});
```

above the ApolloLinkSerialize middleware. It took us a few days to realize what was happening.

Obviously doubling mutations is very very bad. While the user initiated ones are indeed serialized, the doubled ones can land out of order.

**Comment by Francesco-Lanciana** [comment]
Is it that the mutation actually happens twice or it just runs through the links in the pipeline twice? 
Curious to know how you ended up faring with this?

**Comment by dobesv** [comment]
One thing that is strange is that there appears to be a fix for this in the github repo, but the npm package has the bug currently.

@helfer do you think you could push out a release with the latest changes?

**Comment by dobesv** [comment]
I created a new issue about the bad js in the NPM release:

https://github.com/helfer/apollo-link-serialize/issues/14

**Comment by helfer** [comment]
Thanks for opening a ticket @shawnlewis! I wasn't able to reproduce the issue, and from what I could tell there wasn't any difference between the js on npm and the source. Nevertheless, I've published 3.1.2 just in case. Please let me know if this is still a problem.
